x-pack/solutions/observability/plugins/apm/public/components/app/dependencies_inventory/dependencies_inventory_table/index.tsx

```diff
@@ -72,26 +72,29 @@ export function DependenciesInventoryTable() {
     () =>
       data?.dependencies
         ? orderBy(
-            data.dependencies.map((item) => ({
-              name: getNodeName(item.location),
-              impact: item.currentStats.impact,
-              latency: item.currentStats.latency.value,
-              throughput: item.currentStats.throughput.value,
-              failureRate: item.currentStats.errorRate.value,
-            })),
-            sortField,
-            sortDirection
-          )
-            .slice(page * pageSize, (page + 1) * pageSize)
-            .map(({ name }) => name)
-            .sort()
+          data.dependencies.map((item) => ({
+            name: getNodeName(item.location),
+            impact: item.currentStats.impact,
+            latency: item.currentStats.latency.value,
+            throughput: item.currentStats.throughput.value,
+            failureRate: item.currentStats.errorRate.value,
+          })),
+          sortField,
+          sortDirection
+        )
+          .slice(page * pageSize, (page + 1) * pageSize)
+          .map(({ name }) => name)
+          .sort()
         : undefined,
     [data?.dependencies, page, pageSize, sortDirection, sortField]
   );
 
   const { data: timeseriesData, status: timeseriesStatus } = useFetcher(
     (callApmApi) => {
-      if (data?.requestId && visibleDependenciesNames?.length) {
+      if (!visibleDependenciesNames?.length) {
+        return Promise.resolve();
+      }
+      if (data?.requestId) {
```

Comment:
Ideally we want to return Promise resolve when both conditions are false. By keeping the second if in place we could still have the same problem.